packages/node/src/localEvaluation.ts

```diff
@@ -0,0 +1,184 @@
+
+  /**
+   * Construct a local evaluation client.
+   *
+   * Call {@link start} to fetch the flag configs and begin polling for updates,
+   * and {@link evaluate} to evaluate a user for a set of flags.
+   *
+   * @param apiKey The environment's api key.
+   * @param config The {@link LocalEvaluationConfig}.
+   * @param flagConfigCache The cache used to access flag configs to evaluate.
+   */
+  public constructor(
+    apiKey: string,
+    config: LocalEvaluationConfig,
+    flagConfigCache: FlagConfigCache,
+  ) {
+    this.apiKey = apiKey;
+    this.config = { ...LocalEvaluationDefaults, ...config };
+    this.httpClient = FetchHttpClient;
+    this.flagConfigCache = flagConfigCache;
+  }
+
+  /**
+   * Locally evaluates flag variants for a user.
+   *
+   * This function will only evaluated flags for the keys specified in the
+   * {@link flags} argument. If the {@link flags} argument is missing, all flags
+   * in the {@link FlagConfigCache} will be evaluated.
+   *
+   * @param user The user to evaluate
+   * @param flags The flags to evaluate with the user. If empty, all flags from
+   * the flag cache are evaluated.
+   * @returns The evaluated variants
+   */
+  public async evaluate(
+    user: ExperimentUser,
+    flags?: string[],
+  ): Promise<Variants> {
+    // Evaluate the flag configs and user.
+    const flagConfigs = await this.getFlagConfigs(flags);
+    this.debug('[Experiment] evaluate - user:', user, 'flags:', flags);
+    const resultsString = evaluation.evaluate(
+      JSON.stringify(flagConfigs),
+      JSON.stringify(user),
+    );
+    this.debug('[Experiment] evaluate - result:', resultsString);
+    // Parse variant results
+    const results: EvaluationResult = JSON.parse(resultsString);
+    const variants: Variants = {};
+    Object.keys(results).forEach((key) => {
+      variants[key] = {
+        value: results[key].variant.key,
+        payload: results[key].variant.payload,
+      };
+    });
+    return variants;
+  }
+
+  /**
+   * Start polling for flag configurations.
+   *
+   * You must call this function to begin polling for flag config updates.
+   * The promise returned by this function is resolved when the initial call
+   * to fetch the flag configuration completes.
+   *
+   * Calling this function while the poller is already running will stop and
+   * restart the poller.
+   */
+  public async start(): Promise<void> {
+    this.stop();
+    this.debug('[Experiment] poller - start');
+    this.flagConfigPoller = setInterval(async () => {
+      await this.updateFlagConfigs();
+    }, this.config.flagConfigPollingInterval);
+    if (!this.flagConfigPromise) {
+      this.flagConfigPromise = this.updateFlagConfigs({
+        attempts: 5,
+        min: 1,
+        max: 1,
+        scalar: 1,
+      });
+      await this.flagConfigPromise;
+    }
+  }
+
+  /**
+   * Stop polling for flag configurations.
+   *
+   * Calling this function while the poller is not running will do nothing.
```

Comment:
minor nitpick: the words running and scheduled are overloaded. Even though the poller isn't running actively, it can be scheduled and this stop function will stop it from running in the future